src/helpers/generator.ts

```diff
@@ -18,115 +18,153 @@ export const cleanComment = (comment: string): string => {
-          // Capitalize first letter and keep rest as is
-          return cleaned.charAt(0).toUpperCase() + cleaned.slice(1);
-        } catch (err) {
-          console.error(`Error processing part "${part}":`, err);
-          return "";
-        }
-      })
-      .filter(Boolean) // Remove any empty strings from failed processing
-      .join("");
-
-    if (!sanitizedName) {
-      console.warn('sanitizeTypeName: all parts were invalid, using default "UnknownType"');
-      return "UnknownType";
-    }
+interface ParsedSignature {
+  params: string;
+  returnType: string;
+}
 
-    return sanitizedName;
-  } catch (err) {
-    console.error("Error in sanitizeTypeName:", err);
-    return "UnknownType";
+const parseSignature = (signature: string): ParsedSignature | null => {
+  const match = signature?.match(/\((?:this: \w*(?:, )?)?(?<params>.*)\) => (?<returnType>.*)/);
+
+  if (!match?.groups) {
+    return null;
   }
+
+  return {
+    params: match.groups.params,
+    returnType: match.groups.returnType
+  };
 };
 
 export const convertFuncSignatureToType = (
   funcSignature: string,
-  funcType: "methods" | "statics" | "query",
+  funcType: keyof typeof TYPE_MAPPINGS,
   modelName: string
 ): string => {
   try {
-    const sanitizedModelName = sanitizeTypeName(modelName);
+    const sanitizedModelName = sanitizeModelName(modelName);
+    const typeMapping = TYPE_MAPPINGS[funcType](sanitizedModelName);
 
-    // Extract parameters and return type using regex with named groups
-    const signatureMatch = funcSignature?.match(
-      /\((?:this: \w*(?:, )?)?(?<params>.*)\) => (?<returnType>.*)/
-    );
-
-    if (!signatureMatch?.groups) {
-      console.warn(`Invalid function signature format: ${funcSignature}`);
-      return `(this: ${sanitizedModelName}Document) => any`;
+    const parsedSignature = parseSignature(funcSignature);
+    if (!parsedSignature) {
+      console.warn(
+        `Failed to parse function signature: ${funcSignature}, please open an issue if you believe this is an error`
+      );
+      return `(this: ${typeMapping.thisType}) => ${typeMapping.returnType}`;
     }
 
-    const { params, returnType } = signatureMatch.groups;
-    const hasParams = params?.length > 0;
-    const paramsString = hasParams ? `, ${params}` : "";
-
-    // Map function types to their corresponding this types and return types
-    const typeMap = {
-      query: {
-        thisType: `${sanitizedModelName}Query`,
-        returnType: `${sanitizedModelName}Query`
-      },
-      methods: {
-        thisType: `${sanitizedModelName}Document`,
-        returnType: returnType || "any"
-      },
-      statics: {
-        thisType: `${sanitizedModelName}Model`,
-        returnType: returnType || "any"
-      }
-    };
+    const { params, returnType } = parsedSignature;
+    const paramsString = params?.length > 0 ? `, ${params}` : "";
 
-    const { thisType, returnType: mappedReturnType } = typeMap[funcType];
+    const finalReturnType =
```

Comment:
Hmm this new type mapping logic is a bit confusing to me, I think we could clean it up a bit. There should be a way to combine `TYPE_MAPPINGS` and `parseSignature` logic so that we dont need all these conditionals. In general for code like this id prefer more verbose but clear code vs succinct but hard to understand. 

what about something like 

```typescript
const funcTypeToThisSuffix = {
   query: "Query",
   methods: "Document",
  statics: "Model"
}


const parseSignature = (signature: string, modelName: string, funcType: "query" | "methods" | "statics"): ParsedSignature | null => {
  const thisSuffix = funcTypeToThisSuffix[funcType]
  const thisType = `${modelName}${thisSuffix}`
  const queryReturnType = `${modelName}Query`

  const match = signature?.match(/\((?:this: \w*(?:, )?)?(?<params>.*)\) => (?<returnType>.*)/);

  if (!match?.groups) {
    console.warn(`Failed to extract types from function signature: ${funcSignature}, falling back to defaults`);
    const defaultReturnType = funcType === "query" ? queryReturnType : "any"
    const defaultParams = "...args: any[]"

    return {
      params: defaultParams,
      returnType: defaultReturnType,
      thisType
    }
  }

  const returnType = funcType === "query" ? queryReturnType : match.groups.returnType

  return {
    params: match.groups.params,
    returnType: match.groups.returnType,
    thisType
  };
};

```

(im sure this can be improved, and I didn't test it just wrote up quick, so let me know if you have suggestions, but I think this approach is a bit easier to read)

Reply:
yea this is much cleaner. just pushed an update with it included